projects/core/src/lib/layers/symbology/raster-symbology-editor/raster-symbology-editor.component.html

```diff
@@ -1,129 +1,153 @@
+                        >
+                        </geoengine-color-attribute-input>
+                    </div>
 
-            <p fxLayout="row" fxLayoutAlign="space-between center" fxLayoutGap="0.5rem">
-                <span>Colorizer:</span>
-                <mat-button-toggle-group
-                    fxFlex
-                    [multiple]="false"
-                    [ngModel]="getColorizerType()"
-                    (ngModelChange)="updateColorizerType($event)"
-                    [disabled]="getColorizerType() === 'palette'"
-                >
-                    <mat-button-toggle fxFlex value="linearGradient">Linear Gradient</mat-button-toggle>
-                    <mat-button-toggle fxFlex value="logarithmicGradient">Logarithmic Gradient</mat-button-toggle>
-                    <mat-button-toggle fxFlex value="palette" [disabled]="true">Palette</mat-button-toggle>
-                </mat-button-toggle-group>
-            </p>
-            <div
-                *ngIf="symbology.colorizer.isGradient()"
-                class="colorizer-preview"
-                [style.background]="symbology.colorizer | geoengineColorizerCssGradient: 90"
-            ></div>
-        </ng-template>
-    </mat-expansion-panel>
+                    <p fxLayout="row" fxLayoutAlign="space-between center" fxLayoutGap="0.5rem">
+                        <span>Colorizer:</span>
+                        <mat-button-toggle-group
+                            id="gradientType"
+                            fxFlex
+                            [multiple]="false"
+                            [ngModel]="getColorizerType()"
+                            (ngModelChange)="updateColorizerType($event)"
+                            [disabled]="getColorizerType() === 'palette'"
+                        >
+                            <mat-button-toggle fxFlex value="linearGradient">Linear Gradient</mat-button-toggle>
+                            <mat-button-toggle fxFlex value="logarithmicGradient">Logarithmic Gradient</mat-button-toggle>
+                            <mat-button-toggle fxFlex value="palette" [disabled]="true">Palette</mat-button-toggle>
+                        </mat-button-toggle-group>
+                    </p>
+                    <div
+                        *ngIf="symbology.colorizer.isGradient()"
+                        class="colorizer-preview"
+                        [style.background]="symbology.colorizer | geoengineColorizerCssGradient: 90"
+                    ></div>
+                </ng-template>
+            </mat-expansion-panel>
+
+            <mat-expansion-panel expanded="true">
+                <mat-expansion-panel-header>
+                    <mat-panel-title fxLayoutAlign="space-between center" fxLayoutGap="0.5rem">
+                        <mat-icon>looks</mat-icon>
+                        <span fxFlex>Update Color Map</span>
+                    </mat-panel-title>
+                </mat-expansion-panel-header>
```

Comment:
It would be great to only show the colormap creation for the gradients

Reply:
If I am understanding this right, the bottom component "Update Color Map" should be hidden when 'Palette' is selected instead of 'Linear Gradient' or 'Logarithmic Gradient'.
I think this will be easy to achieve with an *ngIf, once Palette mode is implemented. For now, since the `getColorizerType()` method depends on the symbology of the layer being an instance of `PaletteColorizer` (which makes sense of course), I don't think there is a way of returning 'palette' from `getColorizerType()` without an intrusive workaround that would ultimately be reversed later.
So my suggestion would be to wait with this change until the next step, where I'll try to implement the user selected palette

Reply:
`getColorizerType()` seems to already return `palette` as one option. Otherwise, the button wouldn't be selected.

Reply:
In 29ff0dfe42bb1d8e1d1f657b4dfe5934e991ea45 the panel gets hidden when palette is selected

Reply:
![grafik](https://user-images.githubusercontent.com/4947061/205031677-2601f6e4-4aa2-4562-b45b-4c7bd9f96c09.png)

Unfortunately, it is not.

Reply:
Fixed the bug where it's not hidden when Palette was selected as default [3f22461](https://github.com/geo-engine/geoengine-ui/pull/513/commits/3f224612d1c12d14216ac47e94902993d7b3c84d)